src/main/java/opinion/sentences/SentenceParser.java

```diff
@@ -1,25 +1,36 @@
 package opinion.sentences;
 
+import org.jetbrains.annotations.Nullable;
+
 import java.util.Iterator;
 
 class SentenceParser implements Iterator<String> {
 
-    SentenceParser(String strSentence) {
+    private String[] mWords;
+    private int mWordIndex;
+
 
+    public SentenceParser(String strSentence) {
+        mWords = strSentence.trim().split("\\W+"); //
+        mWordIndex = 0;
+        for (int i = 0; i < mWords.length; i++) {
+            // TODO(anna): improve this when it breaks
+            mWords[i] = mWords[i].replaceAll("[^\\w]", "");
+        }
     }
 
+    @Nullable
     @Override
     public String next() {
```

Comment:
This is a public class, so please put javadoc comments on all public methods as well as on the class itself.